### ISSUE TITLE
[BUG] Mailbox::parseDateTime ignores timezones

### ISSUE THREAD

**Issue by HansAdema** [barbushin/php-imap#383]
**Environment (please complete the following information):**
 - PHP IMAP version: 3.0.30
 - PHP Version: 7.3.8
 - Type of execution: CLI

**Describe the bug**

An RFC2822 compliant date string contains a timezone. For an email, this date string is determined by the sending mail server. So when receiving an email, this header can be in any time zone the sending mail server wants it to be in.

My own application imports received emails, and is configured to run in UTC timezone. So I may receive an email with a date like this:

```
Wed, 11 Sep 2019 12:16:09 +0530
```

This time, in UTC, is:

```
2019-09-11 06:46:09
```

However, the `$mail->date` parameter contains the following:

```
2019-09-11 12:16:09
```

As you can see, the time zone was stripped from the string entirely, without any conversion being done. 

Because of this, it's impossible to process this date output as a normalized datetime in UTC.

**To Reproduce**

Simply feed the library a date string in another time zone that what the application is configured to use. The date I provided above is a good example.

**Expected behavior**

I would expect the data parameter to be _something_ I could make sure is in UTC.

Possible solutions include:
- Returning the raw DateTime instance instead of pre formatting it (my preference).
- Returning a string with timezone designation in it.
- Have parseDateTime convert the date string to the local timezone before returning it.

**Screenshots / Outputs**
If applicable, add screenshots or outputs of your script to help explain your problem.

N/A

**Additional context**

N/A

**Comment by Sebi94nbg** [comment]
Thanks for your bug report!

I believe, it makes sense to return a [RFC 3339](https://tools.ietf.org/html/rfc3339) compliant format like `2005-08-15T15:52:01+00:00`, doesn't it?

**Comment by Sebi94nbg** [comment]
Related to #374.

**Comment by Sebi94nbg** [comment]
This library tries to format the given header datetime to a RFC3339 compliant format. If this fails, it will return the original value.

Released in version [3.0.31](https://github.com/barbushin/php-imap/releases/tag/3.0.31).
